lib/auth/saml.go

```diff
@@ -46,6 +46,9 @@ var ErrSAMLNoRoles = trace.AccessDenied("No roles mapped from claims. The mappin
 
 // UpsertSAMLConnector creates or updates a SAML connector.
 func (a *Server) UpsertSAMLConnector(ctx context.Context, connector types.SAMLConnector) error {
+	if err := services.ValidateSAMLConnector(connector, a); err != nil {
```

Comment:
Is this covered by tests? Could we get a simple test case for the bug so we don't regress?

Reply:
I'll see how simple I can make it. I'll need to test that an Upsert with invalid data for an unauthorized user returns an authorization error and not a validation failure error. I will also stand up a test http server and point the entity_descriptor_url to that and ensure it does not get hit in the case of an authorization failure.

I'll also check the code path for OIDC and Github connectors - I have checked that they dont have this same exact issue, but I believe they can have URLs to (maybe that's just OIDC though), so a similar test should be done for them.

Reply:
Thanks for making the effort.